### ISSUE TITLE
Possible Security Problems

### ISSUE THREAD

**Issue by ctindall** [dxctechnology/dxcf-templates#4]
Hey there! I noticed some possible problems in some code in this repo. A quick summary of a few of them is below, but let me know if you're interested in seeing a full report or talking about cloud security in general.

---

**severity:** serious

**filename:** `./Identity-SamlIdentityProvider-Remove.yaml`

**line number(s):** [354, 370]

**resource(s):** 

IAM user should not have any inline policies.  Should be centralized Policy object on group

---

**severity:** warning

**filename:** `./Identity-SamlIdentityProvider-Remove.yaml`

**line number(s):** [220, 389]

**resource(s):** 

IAM role should not allow * resource on its permissions policy

---

**severity:** warning

**filename:** `./Identity-SamlIdentityProvider-Remove.yaml`

**line number(s):** [389]

**resource(s):** 

Resource found with an explicit name, this disallows updates that require replacement of this resource

---

**severity:** warning

**filename:** `./Identity-Groups.yaml`

**line number(s):** [30, 37, 46, 53, 60, 67, 74, 85, 97, 108, 113]

**resource(s):** 

Resource found with an explicit name, this disallows updates that require replacement of this resource

---

**severity:** warning

**filename:** `./Example-PrivateIpAddressCustomResource.yaml`

**line number(s):** [64, 74]

**resource(s):** 

EC2 Subnet should not have MapPublicIpOnLaunch set to true

---

**severity:** warning

**filename:** `./Baseline-Baseline.yaml`

**line number(s):** [161, 311, 446, 567, 662, 787]

**resource(s):** 

IAM role should not allow * resource on its permissions policy

---

**severity:** warning

**filename:** `./Management-VPC.yaml`

**line number(s):** [410, 420, 430, 440, 450, 460]

**resource(s):** 

EC2 Subnet should not have MapPublicIpOnLaunch set to true

**Comment by michael-crawford** [comment]
You seem to be using some automated lint type tool. The issues you mention are not considered by us to be defects - they are part of the design, but I'll address some of them.

- Identity-SamlIdentityProvider-Remove.yaml - no in-line policies. How is this a serious issue? Big applications like CloudCheckr ask you to create users to harvest what they need using in-line policies all the time, and sometimes, when an IAM policy applies specifically to a single user, like for example an Okta integration user, creating a group just to separate IAM policies is more problematic than simply defining the IAM permissions direct on the user.
- Baseline-Baseline.yaml and more - IAM role should not allow * resource on its permissions policy. This is just wrong. Many IAM policies do not allow specific resources, and you MUST use '*'. At other times, you want the policy to apply to every resource. In this specific case, we are listing Lambda Layers, which must allow listing all of them for it to work. Ditto setting up an IAM policy to have a user list their own bucket contents - they can't get to their own bucket in the console unless they can list all other buckets. Many other examples like this. Tools like whatever you used can help you find issues, but there are always false positives - issues found which are not actually a problem, often are correct, such as the case here.
- Management-VPC and more - EC2 Subnet should not have MapPublicIpOnLaunch set to true. First time I've heard this, and I disagree. We create subnets specifically called PublicSubnetA, PublicSubnetB, and only on such subnets do we set this value to true, because we want this to be true. We only deploy instances into public subnets when we want to reach them via a public IP. Even if the public at large can't get to the server, often we want our administrators to reach them direct from some CIDRs if they are acting up. We use Security Group Ingress rules to define such access. No access is allowed without such rules. I can't see why this even qualifies as a warning.

In summary, not one of the issues you've pointed out is something we don't know or think is incorrect. So, closing this thread.

If you have more substantive comments based on thoughtful analysis, I'd love to hear them. But please don't send me the results of running some lint tool thinking that's going to impress me. it won't, without additional comment as to why you really think there's a problem.
